File: 1930-unique-length-3-palindromic-subsequences/1930-unique-length-3-palindromic-subsequences.js
/**
 * @param {string} s
 * @return {number}
 */
var countPalindromicSubsequence = function(s) {
    let position = new Map();
    for (let i = 0; i < s.length; i++){
        if (position.has(s[i])){
            position.get(s[i])[1] = i;
        }
        else{
            position.set(s[i], [i, i]);
        }
    }
    
    let result = 0;
    for (let [left, right] of position.values()){
        let count = new Set();
        for (let i = left+1; i < right; i++){
            count.add(s[i]);
        }
        result += count.size;
    }
    
    return result;
};

File: 1930-unique-length-3-palindromic-subsequences/1930-unique-length-3-palindromic-subsequences.py
class Solution:
    def countPalindromicSubsequence(self, s: str) -> int:
        position = {}
        
        for i, c in enumerate(s):
            if c in position:
                position[c][1] = i
            else:
                position[c] = [i, i]
                
        result = 0
        for i, j in position.values():
            count = set()
            for k in range(i+1, j):
                count.add(s[k])
            result += len(count)
        
        return result